NUModules/NUModule.j

```diff
@@ -2077,9 +2077,34 @@ NUModuleTabViewModeIcon                = 2;
 }
 
 /*! Apply the advanced given filers.
-*/
-- (void)applyAdvancedFilters:(CPString)aString
-{
+Date filters will return the default date toString in format Mon May 21 2018 10:59:06 GMT-0700 (PDT)
+We need to intercept predicates and translate dates to format: MM/DD/YYYY HH:MIN:SS timezoneoffset
+*/
+- (void)applyAdvancedFilters:(CPPredicate)aPredicate
+{
+    var subpredicates = [aPredicate subpredicates];
+    var isModified = false;
+    var subPredicates = [];
+    for (var i = 0; i < [subpredicates count]; i++) {
+      var predicate = subpredicates[i];
+      var lhs = [predicate leftExpression];
+      var valueForKeyPath = [[predicate rightExpression] expressionValueWithObject:nil context:nil];
+      var date = new Date(valueForKeyPath);
```

Comment:
how do we assume `valueForKeyPath` corresponds to a date? will `[subpredicates count] > 0` be satisfied only for date?

Reply:
Not really! Date.parse will validate for a valid date format. This is from MDN site:
```
A number representing the milliseconds elapsed since January 1, 1970, 00:00:00 UTC and the date obtained by parsing the given string representation of a date. If the argument doesn't represent a valid date, NaN is returned.
```